### PR TITLE
Support ENV variables in CRDs

### PR DESCRIPTION
**Pull request by hossainemruz**
Fixes: https://github.com/kubedb/project/issues/163

Task List:
- [x] Added support for ENV variable
- [x] Added tests
- [x] Update doc

### CODE REVIEW

File: hack/docker/postgres/10.2/scripts/primary/start.sh [hunk 1]
```diff
@@ -22,13 +34,38 @@ mv /tmp/postgresql.conf "$PGDATA/postgresql.conf"
...
+    op="CREATE"
+fi
+
+WITH_POSTGRES_PASSWORD="PASSWORD '$POSTGRES_PASSWORD'"
+
 # alter postgres superuser
-psql --username postgres <<-EOSQL
-ALTER USER postgres WITH SUPERUSER PASSWORD '$PGPASSWORD';
+"${psql[@]}" --username postgres <<-EOSQL
+    $op USER "$POSTGRES_USER" WITH SUPERUSER  $WITH_POSTGRES_PASSWORD;
```
Comment (the-redback) [inline hack/docker/postgres/10.2/scripts/primary/start.sh]: Try to skip using this variable `WITH_POSTGRES_PASSWORD` (as the parameter is not redundant) and use that parameter here.

File: hack/docker/postgres/9.6.7/scripts/primary/start.sh [hunk 2]
```diff
@@ -22,13 +35,38 @@ mv /tmp/postgresql.conf "$PGDATA/postgresql.conf"
...
+    echo
+fi
+
+if [ "$POSTGRES_USER" = "postgres" ]; then
+    op="ALTER"
+else
+    op="CREATE"
+fi
+
+WITH_POSTGRES_PASSWORD="PASSWORD '$POSTGRES_PASSWORD'"
```
Comment (the-redback) [inline hack/docker/postgres/9.6.7/scripts/primary/start.sh]: same here

File: hack/docker/postgres/10.2/scripts/primary/start.sh (hunk 1 above)
Comment (hossainemruz) [reply hack/docker/postgres/10.2/scripts/primary/start.sh]: done

File: hack/docker/postgres/9.6.7/scripts/primary/start.sh (hunk 2 above)
Comment (hossainemruz) [reply hack/docker/postgres/9.6.7/scripts/primary/start.sh]: done

**Pull request merged**